lib/src/pgn.dart

```diff
@@ -0,0 +1,1006 @@
+                    token == '*') {
+                  if (_stack.length == 1 && token != '*') {
+                    _gameHeaders['Result'] = token;
+                  }
+                } else if (token == '(') {
+                  _stack.add(_ParserFrame(parent: frame.parent, root: false));
+                } else if (token == ')') {
+                  if (_stack.length > 1) _stack.removeLast();
+                } else if (token == '{') {
+                  final openIndex = match.end;
+                  final beginIndex =
+                      line[openIndex] == ' ' ? openIndex + 1 : openIndex;
+                  line = line.substring(beginIndex);
+                  _state = ParserState.comment;
+                  continue continuedLine;
+                } else {
+                  if (token == 'Z0' || token == '0000' || token == '@@@@') {
+                    token = '--';
+                  } else if (token.startsWith('0')) {
+                    token = token.replaceAll('0', 'O');
+                  }
+                  if (frame.node != null) {
+                    frame.parent = frame.node!;
+                  }
+                  frame.node = ChildNode(PgnNodeData(
+                      san: token, startingComments: frame.startingComments));
+                  frame.startingComments = null;
+                  frame.root = false;
+                  frame.parent.children.add(frame.node!);
+                }
+              }
+            }
+            return;
+          }
+
+        case ParserState.comment:
+          {
+            final closeIndex = line.indexOf('}');
+            if (closeIndex == -1) {
+              _commentBuf.add(line);
+              return;
+            } else {
+              final endIndex = closeIndex > 0 && line[closeIndex - 1] == ' '
+                  ? closeIndex - 1
+                  : closeIndex;
+              _commentBuf.add(line.substring(0, endIndex));
+              _handleComment();
+              line = line.substring(closeIndex);
+              _state = ParserState.moves;
+              freshLine = false;
+            }
+          }
+      }
+    }
+  }
+
+  void _handleNag(int nag) {
+    final frame = _stack[_stack.length - 1];
+    if (frame.node != null) {
+      frame.node!.data = frame.node!.data.copyWith(nag: nag);
+    }
+  }
+
+  void _handleComment() {
+    final frame = _stack[_stack.length - 1];
+    final comment = _commentBuf.join('\n');
+    _commentBuf = [];
+    if (frame.node != null) {
+      frame.node!.data = frame.node!.data.copyWith(comment: comment);
+    } else if (frame.root) {
+      _gameComments.add(comment);
+    } else {
+      frame.startingComments ??= [];
+      frame.startingComments!.add(comment);
+    }
+  }
+}
+
+/// Default function to parse a PGN
+///
+/// Creates a list of games if multiple games found
+/// Provide a optional function [initHeaders] to create different headers other than the default
+List<Game<PgnNodeData>> parsePgn(String pgn,
```

Comment:
let's keep this function to have the list of game parsing feature but let's rename it to `parseMultiGamePgn` for clarity.

I'd like to have a `factory PgnGame.parsePgn` constructor to return a single `PgnGame` as well.

Reply:
I could not create a factory constructor `PgnGame.parsePgn` as `PgnGame` takes a generic type, so its constructor cannot return a `PgnGame<PgnNodeData>`. 
Implemented a static method instead.